src/TextDateOfBirthInput/TextDateOfBirthInput.tsx

```diff
@@ -4,25 +4,26 @@ import styled from 'styled-components'
 import { Box } from '../Box'
 import { Fieldset, FieldsetProps } from '../fields/Fieldset'
 import { TextInput } from '../TextInput'
+import { InputValidationError } from '../utils/dateOfBirth/schema'
+import { FieldError } from 'react-hook-form'
 
-export interface DateObjectType {
-  day: string
-  month: string
-  year: string
+export interface CompleteDateObject {
```

Comment:
Nitpick: Is this type even used now that we have `DateObject`?

I think if you are not using it in cms then you can either:
- delete it and make 
```
export type DateObject = {
  day?: string | null
  month?: string | null
  year?: string | null
}
```
- or you can just keep each key of `CompleteDateObject` type as non-undefined as the DateObject makes them optional anyways `[K in keyof CompleteDateObject]?:`